vike/client/client-routing-runtime/prefetch.ts

```diff
@@ -41,6 +68,42 @@ async function prefetchAssets(pageId: string, pageContext: PageContextUserFiles)
   }
 }
 
+async function prefetchPageContext(
+  pageId: string,
+  pageContext: {
+    urlOriginal: string
+    _objectCreatedByVike: boolean
+    _urlHandler: null
+    _urlRewrite: null
+    _baseServer: string
+    _pageFilesAll: PageFile[]
+    _pageConfigs: PageConfigRuntime[]
+    _pageConfigGlobal: PageConfigGlobalRuntime
+    _allPageIds: string[]
+    _pageRoutes: PageRoutes
+    _onBeforeRouteHook: Hook | null
+  } & PageContextUrlClient
+): Promise<void> {
+  try {
+    objectAssign(
+      pageContext,
+      await loadUserFilesClientSide(pageId, pageContext._pageFilesAll, pageContext._pageConfigs)
+    )
+    objectAssign(pageContext, {
+      isHydration: false as const,
+      isBackwardNavigation: null,
+      _hasPageContextFromServer: false as const,
+      _hasPageContextFromClient: true as const,
+      _pageId: pageId,
+      _pageConfigs: pageContext._pageConfigs
+    })
+    const res = await getPageContextFromHooks_isNotHydration(pageContext, false)
```

Comment:
Actually, how about we use `fetchPageContextFromServer()` instead of `getPageContextFromHooks_isNotHydration()`? I think it would be more precise (avoids calling client-side hooks), and also maybe we can then skip the need to call `loadUserFilesClientSide()`.

Reply:
You're right, it requires a little bit of refactoring. But I think it's worth it: I think it's more accurate/correct to avoid calling client-side hooks upon prefetching. Maybe the solution is to split `getPageContextFromHooks_isNotHydration()` in two parts?

Reply:
All right👍
I split `getPageContextFromHooks_isNotHydration()` into two parts.